_posts/en/newsletters/2019-02-26-newsletter.md

```diff
@@ -0,0 +1,152 @@
+---
+title: 'Bitcoin Optech Newsletter #35'
+permalink: /en/newsletters/2019/02/26/
+name: 2019-02-26-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes the availability of a libsecp256k1 fork
+implementing BIP-Schnorr compatible signatures, lists popular questions
+and answers for February from the Bitcoin StackExchange, and describes
+notable merges in popular Bitcoin infrastructure projects.
+
+## Action items
+
+None this week.
+
+## News
+
+- **Schnorr-ready fork of libsecp256k1 available:** Blockstream
+  cryptographer Andrew Poelstra [announced][schnorr libsecp256k1-zkp]
+  that the [libsecp256k1-zkp][] library used in [Elements Project][]
+  based sidechains (such as [Liquid][]) now supports [BIP-Schnorr][]
+  compatible signatures in a variety of configurations:
+
+    - Basic single pubkeys and signatures.  These are almost identical
+      in use to those with Bitcoin's current ECDSA algorithm, although
+      signatures are serialized to use about eight fewer bytes and can
+      be efficiently verified in batches.
+
+    - [MuSig][] for multiparty signatures.  Onchain, these look
+      identical to single pubkeys and signatures but the public keys and
+      signatures are generated by a set of tools that each bring their
+      own private keys and cooperate in a multi-step protocol.  Whereas
+      multisig using current Bitcoin Script requires up to *n* pubkeys
+      and signatures for k-of-n multisig security, MuSig can provide the
+      same security using just one public key and one signature---reducing
+      block chain space, improving verification efficiency, increasing
+      privacy, and allowing much larger sets of signers than supported
+      by Bitcoin Script's current byte-size and signature-operation
+      limits.  However, the multi-step protocol requires especially
+      careful management of secret nonces; Poelstra's post details
+      how libsecp256k1-zkp attempts to minimize the risk of
+      nonce-related failures and teases the possibility of even better
+      solutions in the future.
```

Comment:
I'm not sure you want to cover this level of detail, but another point you could make is that there are a few approaches to using Musig:
1) Threshold signatures, as implied in your text above, but you might want to point out that a drawback with this approach is loss of accountability. If the signing parties care about public or private auditing, this could be a non-starter for them.
2) It is possible to achieve k of n multisig with accountability and still get the benefits of 1 sig/1 pubkey for the direct spending case (ie most common signing scenario) if Taproot is used. It can be structured in a couple of different ways:
a) If there is a clear spending case that is most likely and most convenient, such as 2 online keys in a 2of3 mulitsig hot wallet, then the Taproot direct spending key can be the threshold 2 online keys, and the keytree can include the other spending options (such as the 3rd key, perhaps in a cold wallet) paired with either of the online keys. While this configuration gives you accountability, the drawback is that in the scenarios where the cold wallet key is used, you would lose the blockchain efficiency, privacy, and fungibility benefits as the spending script would be part of the transaction.
b) If all n keys of a k of n are all online, making it relatively convenient to do the interactive signing protocol for any of the n keys, then it could make sense to structure the direct spending key as an n of n key, with all k pairing signing configurations as part of the keytree. In this scenario, any k signing participants could prove to the other n - k parties that they have sufficient signing power to approve spending the funds, so the other n - k parties might as well sign the n of n direct spending key, with the benefit being that in any signing scenario there would only be 1 signature and 1 pubkey broadcast. The accountability in this scenario would be private and not public, although it could be made public if it were important for the use case.

Reply:
Re: point 1, are you sure there's a loss of "private auditing"?  Can't the subset of participating signers publish the transcript of the data exchanged during signature generation?  (Alternatively, can't participating signers each commit to their intention to sign before starting and then commit to that set of commitments in signature, e.g. using sign-to-contract?)

Reply:
Ignore my previous comment.  After sleeping on it, I realized the methods I describe only work in the honest case; in the pathological case, the colluding signers can spend the money without generating an audit trail.  Accountability that only works in the honest case is pretty useless.  :-)

I'll try to work a brief summary of your concern in point 1 into the newsletter description.  Also your mention of taproot reminds me that Poelstra's blog post doesn't itself mention creating or verifying the pay-to-contract pubkeys used for taproot; I need to check to see if the library supports those directly, which might also be worthy of a mention.  (I know Blockstream has code for that somewhere, since it's integral to the Elements design and I've used it myself via their old [contracthash](https://github.com/Blockstream/contracthashtool) tool.)  I'm probably not going to mention your point 2 workarounds, which are clever but also IMO something not directly related to the functions being made available in libsecp256k1-zkp (but I'll definitely keep them in mind if we cover threshold signing in more detail in the future).

Thanks for teaching me something!